Dockerfile.konflux

```diff
@@ -0,0 +1,64 @@
+ARG DEV_IMAGE=registry.redhat.io/ubi9/go-toolset@sha256:fd41c001abc243076cc28b63c409ae6d9cbcad401c8124fb67d20fe57a2aa63a
```

Comment:
can't we call this file as Containerfile?
It is better than dockerfile, and it is aligned with Podman.

Reply:
We trying to keep standardized naming across all components as Dockerfile.konflux.